vcd/resource_vcd_org_vdc.go

```diff
@@ -263,17 +297,80 @@ func resourceVcdVdcRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorRetrievingOrg, err)
 	}
 
-	vdc, err := adminOrg.GetVdcByName(d.Id())
-	if err != nil || vdc == (govcd.Vdc{}) {
-		log.Printf("[DEBUG] Unable to find vdc. Removing from tfstate")
-		d.SetId("")
-		return nil
+	adminVdc, err := adminOrg.GetAdminVdcByName(d.Get("name").(string))
+	if err != nil || adminVdc == (govcd.AdminVdc{}) {
+		log.Printf("[DEBUG] Unable to find vdc")
+		return fmt.Errorf("unable to find vDC %#v", err)
+	}
+
+	d.Set("allocation_model", adminVdc.AdminVdc.AllocationModel)
+	d.Set("cpu_guaranteed", strconv.FormatFloat(*adminVdc.AdminVdc.ResourceGuaranteedCpu, 'f', 2, 64))
+	d.Set("cpu_speed", adminVdc.AdminVdc.VCpuInMhz)
+	d.Set("description", adminVdc.AdminVdc.Description)
+	d.Set("enable_fast_provisioning", adminVdc.AdminVdc.UsesFastProvisioning)
+	d.Set("enable_thin_provisioning", adminVdc.AdminVdc.IsThinProvision)
+	d.Set("enable_vm_discovery", adminVdc.AdminVdc.VmDiscoveryEnabled)
+	d.Set("enabled", adminVdc.AdminVdc.IsEnabled)
+	d.Set("memory_guaranteed", strconv.FormatFloat(*adminVdc.AdminVdc.ResourceGuaranteedMemory, 'f', 2, 64))
+	d.Set("name", adminVdc.AdminVdc.Name)
+
+	networkPool, err := govcd.GetNetworkPoolByHREF(vcdClient.VCDClient, adminVdc.AdminVdc.NetworkPoolReference.HREF)
+	if err != nil {
+		return fmt.Errorf("error retrieving network pool: %#v", err)
 	}
 
-	log.Printf("[TRACE] vdc read completed: %#v", vdc.Vdc)
+	d.Set("network_pool_name", networkPool.Name)
+	d.Set("network_quota", adminVdc.AdminVdc.NetworkQuota)
+	d.Set("nic_quota", adminVdc.AdminVdc.Vdc.NicQuota)
+	d.Set("provider_vdc_name", adminVdc.AdminVdc.ProviderVdcReference.Name)
+	d.Set("vm_quota", adminVdc.AdminVdc.Vdc.VMQuota)
+
+	d.Set("compute_capacity", adminVdc.AdminVdc.ComputeCapacity)
+
+	d.Set("storage_profile", adminVdc.AdminVdc.VdcStorageProfiles)
+
+	log.Printf("[TRACE] vdc read completed: %#v", adminVdc.AdminVdc)
 	return nil
 }
 
+func resourceVcdVdcUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] vdc update initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrg, err)
+	}
+
+	vdcName := d.Get("name").(string)
+	if d.HasChange("name") {
+		oldValue, _ := d.GetChange("name")
+		vdcName = oldValue.(string)
+	}
+
+	adminVdc, err := adminOrg.GetAdminVdcByName(vdcName)
+	if err != nil || adminVdc == (govcd.AdminVdc{}) {
+		log.Printf("[DEBUG] Unable to find vDC.")
```

Comment:
"vDC" -> "VDC"  (also in the rest of the PR)

Reply:
Fixed through file